common/expbloom/bloom.go

```diff
@@ -0,0 +1,101 @@
+package bloom
+
+import (
+	"hash"
+	"sync"
+	"time"
+
+	bloomfilter "github.com/holiman/bloomfilter/v2"
+)
+
+const (
+	k = 4
+)
+
+type ExpiringBloom struct {
+	currentBloom int
+	union        *bloomfilter.Filter
+	blooms       []*bloomfilter.Filter
+	size         uint64
+
+	timer *time.Ticker
+	// Mutex lock the currentBloom and union variables
+	mu      sync.RWMutex
+	closeCh chan struct{}
+}
+
+func NewExpiringBloom(n, m uint64, timeout time.Duration) (*ExpiringBloom, error) {
```

Comment:
```suggestion
// NewExpiringBloom creates a new expiring bloom filter, which aggregates
// a series of internal bloom filters. 
//
// m is the size of the internal Bloom filters (in bytes)
// n is the number of filters
// Each internal bloom filter will use 4 keys (the constant `k`). 
// The total size consumption is (on the order of ) m * n bytes.
func NewExpiringBloom(n, m uint64, timeout time.Duration) (*ExpiringBloom, error) {
```
If I calculate correctly, each peer will use `420KB` for these, so nearly half a MB per peer. I guess that is reasonable